src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1827,6 +1798,29 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 }
                 break;
 
+            case NI_Vector64_GetElement:
+            case NI_Vector128_GetElement:
+            {
+                assert(hasImmediateOperand);
+                assert(varTypeIsIntegral(intrin.op2));
+
+                if (intrin.op2->IsCnsIntOrI())
+                {
+                    MakeSrcContained(node, intrin.op2);
+                }
+
+                if (IsContainableMemoryOp(intrin.op1))
```

Comment:
It can be a `GT_IND`, `GT_LCL_FIELD`, or a non-enregisterable `GT_LCL_VAR`.

The JIT will simply generate a direct load of the relevant element. So if you had a `Vector128<float>` and you were accessing `index 3` it would simply generate something like:
`vmovss xmm0, [rdx+0xc]`